sklearn/metrics/classification.py

```diff
@@ -1589,11 +1593,14 @@ def log_loss(y_true, y_pred, eps=1e-15, normalize=True, sample_weight=None):
     The logarithm used is the natural logarithm (base-e).
     """
     lb = LabelBinarizer()
-    T = lb.fit_transform(y_true)
+    lb.fit(labels) if labels is not None else lb.fit(y_true)
+
+    T = lb.transform(y_true)
+    
     if T.shape[1] == 1:
```

Comment:
So I'd say "if T.shape == 1 and len(labels) == 2". Is there a check that the shape of y_pred matches len(labels)?